spec/requests/memberships_controller_spec.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+describe MembershipsController do
+  describe '#create' do
+    context 'when user not a member' do
+      let(:user) { create(:user) }
+      let(:group) { create(:group) }
+      before(:each) { login_as(user) }
```

Comment:
Here you have one `it` block in the `context` group, it's more advised to move these to there. Using `let`/`before` blocks can be very helpful, but in this case it just increases the complexity. I'd recommend
```ruby
it 'creates membership' do
  user  = create(:user)
  group = create(:group)
  login_as(user)
  ...
```

Other solution could be if you put the all of the `let`  blocks in the most outer `describe` block and you just define the differences in the `context` blocks if necessary, but in this case I recommend the first one.

Also the same applies to the further specs in this file.

Reply:
If I move all `let`/`before` blocks to the `it` block, should I remove the enclosing  `context` block? It doesn't do anything except appending a string to the tests name.

Reply:
I’d say no. It helps the understanding and moving everything to the `it` block would only break the flow of the spec. Adding some blocks wouldn’t change the time necessary for running the tests, but can be very helpful when someone tries to understand it. For example:

```ruby
describe MembershipController do
  describe '#create' do
    context 'when the user is not member of the group' do
      it 'creates the membership' do
```